docker-installer/install_cni_plugins.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+set -e
+
+if [ $# -ne 2 ]; then
+  echo "USAGE: $0 source-dir dest-dir"
+  exit 1
+fi
+
+SRC=$1
+DST=$2
+
+install_cni_plugin() {
+	NAME=$1
+	MD5=$( md5sum $SRC/$NAME | awk '{ print $1 }' )
+	if [ -e $DST/$NAME ]; then
+		if [ ! -e $DST/$NAME.by_cni_installer_image ]; then
+			# The file already exists but there's no marker that
+			# it was installed by this installer -> keep untouched
+			echo "* '$NAME' ignored (exists but not installed by me)"
+			return
+		fi
+		OTHER_MD5=$( md5sum $DST/$NAME | awk '{ print $1 }' )
+		INSTALLED_MD5=$( cat $DST/$NAME.by_cni_installer_image )
+
+		if [ "$OTHER_MD5" != "$INSTALLED_MD5" ]; then
+			# The file was previously installed by this installer
+			# but later changed -> keep untouched
+			echo "* '$NAME' ignored (previously installed by me but changed by someone else)"
+			return
+		fi
+
+		if [ "$OTHER_MD5" == "$MD5" ]; then
+			# The file was previously installed by this installer
+			# and is up-to-date
+			echo "* '$NAME' is up-to-date"
+			return
+		fi
+
+		# The file was previously installed by this installer
+		# but needs an update
+		cp -a $SRC/$NAME $DST/$NAME
```

Comment:
Can you use atomic copies here and below? Basically, it looks something like this
```shell
cp -a $SRC/$NAME $DST/.$NAME.new
mv $DST/.$NAME.new $DST/$NAME
```
This prevents the (unlikely but possible) case where the CNI binary is partially written while it executes, as well as attempting to write while the binary is in use.

Reply:
yes, i noticed that `calico` did the same: https://github.com/projectcalico/cni-plugin/blob/v3.15.5/k8s-install/scripts/install-cni.sh#L92-L96